### ISSUE TITLE
[Compaction] Tablet of a static dataset can't be fully compacted into one rowset

### ISSUE THREAD

**Issue by gaodayue** [apache/incubator-doris#2668]
**Describe the bug**
I loaded one static dataset into a Doris table. I expect all tablets to be compacted into one rowset after some time since there is no more new data. But it turns out all tablets end up having 2 or 3 versions.

```
$ curl 'http://$be:$port/api/compaction/show?tablet_id=12492&schema_hash=1862098505'
{
    "cumulative point": 229,
    "last cumulative failure time": "2020-01-06 16:39:05.724",
    "last base failure time": "2020-01-06 14:45:37.679",
    "last cumulative success time": "2020-01-02 18:37:35.086",
    "last base success time": "2020-01-02 19:01:21.834",
    "versions": [
        "[0-228] 6 ",
        "[229-229] 1 ",
        "[230-230] 1 "
    ]
}
$ curl 'http://$be:$port/api/compaction/show?tablet_id=12508&schema_hash=1862098505'
{
    "cumulative point": 230,
    "last cumulative failure time": "2020-01-06 17:56:42.175",
    "last base failure time": "2020-01-06 17:56:45.007",
    "last cumulative success time": "2020-01-02 19:24:21.047",
    "last base success time": "2020-01-02 19:32:26.552",
    "versions": [
        "[0-229] 12 ",
        "[230-230] 2 "
    ]
}
``` 

The reason I want all tablets to be base-compacted into one version is that the aggregated table has a long varchar key column which cause the merge_iterator to be slow, union_iterator can be utilized to boost query performance when there is only one version.

**Environment**
Latest master branch with default configurations.

**Comment by gaodayue** [comment]
Related to #2551 and hopefully solved by #2558 , will verify later

**Comment by gaodayue** [comment]
Fixed by #2558 and #2693
